astropy/utils/tests/test_data.py

```diff
@@ -645,6 +647,28 @@ def test_download_cache():
     assert not os.path.isdir(lockdir), "Cache dir lock was not released!"
 
 
+@pytest.mark.remote_data(source="astropy")
+def test_download_certificate_verification_failed():
+    """Tests for https://github.com/astropy/astropy/pull/10434"""
+
+    # First test the expected exception when download fails due to a
+    # certificate verification error; we simulate this by passing a bogus
+    # CA directory to the ssl_context argument
+    ssl_context = {'cafile': None, 'capath': '/does/not/exist'}
```

Comment:
I thought the `/` in path would make test fail on Windows because it might raise an error about the path instead of URL, but I tested this and it passed.